### DISCUSSION TITLE
configure Consul access with both Secret Store token and consul's bootstrap acl token

### DISCUSSION THREAD

**Discussion by dipshirajput** [edgexfoundry/community#67] [Q&A]
Hi All,
while I am trying to run EdgeX services along with vault and consul using Docker containers, I am facing an error of "**failed to configure consul access**" in consul log where consul container tries to "`**configure Consul access with both Secret Store token and consul's bootstrap acl token**`" with an error of unable to hit http://edgex-vault:8200/ and throwing 403 error .
but this error is inconsistent. sometimes it is successfully able to hit that url and configure consul access, but sometimes it is failing to do so.
attached below is the consul log,
![Screenshot from 2023-03-07 15-05-37](https://user-images.githubusercontent.com/52161957/223444078-0201df66-cb73-4a87-a561-a1f213ccb32f.png)

attached docker-compose file being used,
[docker-compose.txt](https://github.com/edgexfoundry/community/files/10910119/docker-compose.txt)

**Note:**
have created docker images for riscv platform and using jakarta version.

please let me know what could be the probable issue because of which the consul container is experiencing such inconsistent behaviour.
please let me know for any suggestions . Thanks in advance.

**Comment by bnevis-i** [comment]
The snap mechanism uses systemd sequencing and the docker mechanism uses a TCP socket stage-gating mechanism to make sure early services come up in the right order.  Vault needs to come up first, following by security-secretstore-setup, followed by consul.  Consul depends on the Vault token created by security-secretstore-setup to configure Vault's consul secrets engine (this service hands out consul tokens to other edgex services).  Consul is a bit wierd in that it starts up initially in a security-disabled mode, then we reconfigure it to enable consul ACL mode, restart it, and then configure Vault.  There looks like there may also be some oddities in that flow as well (consul/data/acl-bootstrap-reset).  Make sure that you aren't losing any persistent state, and that consul is bootstrapped within an hour of running secretstore-setup (the vault token expire if not used).

**Reply by dipshirajput** [reply]
Thank you @bnevis-i . its resolved.

**Reply by bnevis-i** [reply]
@dipshirajput, these discussions should be searchable on search engines.  It would be helpful to other users who come across your issue in a search if you generously posted what ultimately fixed your issue.
